tests/typescript/conformance/types/alias/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,24 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`aliasArrow.ts 1`] = `
+type simple0 = ()=>string;
+type simple1 = ( ()=>string );
+type simple2 = ((x:number)=>string);
+
+type f0 = number | (() => string);
+type f1 = () => string | number;
+type f2 = (() => string) | number;
+type f3 = (() => string) | (() => boolean);
+type f4 = (() => string) & (() => boolean);
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+type simple0 = () => string;
+type simple1 = () => string;
+type simple2 = (x: number) => string;
+
+type f0 = number | (() => string);
+type f1 = () => string | number;
+type f2 = (() => string) | number;
```

Comment:
@azz 
should we add parens around function type in `f1`?

Reply:
```ts
() => string | number;
// is actually
() => (string | number);
```

Reply:
@azz, sorry for this delayed response. (I have no too much time at recently.

I just want to strip the parens around function type at first.
But it's more complicated than what my expected, 😂.

Is there any other case I should consider？
(Maybe someone else would like to take over this PR if I don't have time?